game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/FirstStrikeStepOrderTest.java

```diff
@@ -0,0 +1,531 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class FirstStrikeStepOrderTest {
+
+  @ParameterizedTest
+  @MethodSource
+  void getStep(final List<BattleStateVariation> parameters, final List<FirstStrikeStepOrder> expected, final List<FirstStrikeStepOrder> expected2) {
+
+    final BattleState battleState = givenBattleState(parameters);
+
+    final List<FirstStrikeStepOrder> steps = FirstStrikeStepOrder.calculate(battleState);
+
+    if (expected.isEmpty()) {
+      assertThat(steps, hasSize(0));
+    } else {
+      System.out.println(steps);
+      assertThat(steps, contains(expected.toArray()));
+    }
+
+    final List<FirstStrikeStepOrder> steps2 = FirstStrikeStepOrder.calculate2(battleState);
+    if (expected2.isEmpty()) {
+      assertThat(steps2, hasSize(0));
+    } else {
+      System.out.println(steps2);
+      assertThat(steps2, contains(expected2.toArray()));
+    }
+  }
+
+  enum BattleStateVariation {
+    HAS_ATTACKING_FIRST_STRIKE,
+    HAS_DEFENDING_FIRST_STRIKE,
+    HAS_ATTACKING_DESTROYER,
+    HAS_DEFENDING_DESTROYER,
+    HAS_WW2V2,
+    HAS_DEFENDING_SUBS_SNEAK_ATTACK
+  }
+
+  BattleState givenBattleState(final List<BattleStateVariation> parameters) {
+    final List<Unit> attackingUnits = new ArrayList<>();
+    attackingUnits.add(givenAnyUnit());
+    if (parameters.contains(BattleStateVariation.HAS_ATTACKING_DESTROYER)) {
+      attackingUnits.add(givenUnitDestroyer());
+    }
+    if (parameters.contains(BattleStateVariation.HAS_ATTACKING_FIRST_STRIKE)) {
+      attackingUnits.add(givenUnitFirstStrike());
+    }
+    final List<Unit> defendingUnits = new ArrayList<>();
+    defendingUnits.add(givenAnyUnit());
+    if (parameters.contains(BattleStateVariation.HAS_DEFENDING_DESTROYER)) {
+      defendingUnits.add(givenUnitDestroyer());
+    }
+    if (parameters.contains(BattleStateVariation.HAS_DEFENDING_FIRST_STRIKE)) {
+      defendingUnits.add(givenUnitFirstStrike());
+    }
+    final MockGameData gameData = MockGameData.givenGameData();
+
+    if (parameters.contains(BattleStateVariation.HAS_DEFENDING_FIRST_STRIKE)
+        && !parameters.contains(BattleStateVariation.HAS_ATTACKING_FIRST_STRIKE)) {
+      gameData.withDefendingSuicideAndMunitionUnitsDoNotFire(false);
+    }
+
+    if (parameters.contains(BattleStateVariation.HAS_DEFENDING_SUBS_SNEAK_ATTACK)) {
+      if (!parameters.contains(BattleStateVariation.HAS_WW2V2)) {
+        gameData.withWW2V2(false);
+      }
+      gameData.withDefendingSubsSneakAttack(true);
+    }
+    if (parameters.contains(BattleStateVariation.HAS_WW2V2)) {
+      gameData.withWW2V2(true);
+    }
+
+    return givenBattleStateBuilder()
+        .attackingUnits(attackingUnits)
+        .defendingUnits(defendingUnits)
+        .gameData(gameData.build())
+        .build();
+  }
+
+  static List<Arguments> getStep() {
+    return List.of(
+        //1
```

Comment:
nit, the numbering is a smell/maintenance problem.

It's a smell as it should be pretty easy to find which parameterized test case is failing. The maintenance burden comes in though if we insert an element mid-list, then suddenly you have to renumber them all.

For now this is okay IMO, but if there is a good way to avoid it, we should do that